### PR TITLE
Fixed typing for middleware functions

### PR DESCRIPTION
**Pull request by sanderphilipse**

### CODE REVIEW

**Comment by muratcorlu** [comment]
Thanks for your contribution!

OK, you directly use Express’s `Request` and `Response` types. That’s a practical solution but lambda-expressless doesn’t cover all of the functionalities of Express Request and Response objects yet. And some of the methods have small behaviour differences as I mentioned in Readme file. But the target is to have Request and Response objects as same as ExpressJS. So, I'm not sure about to continue with ExpressJS types directly or try to keep our own custom types for a while.

Do you have a comment about this? @sanderphilipse

**Comment by cenkingunlugu** [comment]
> Thanks for your contribution!
> 
> OK, you directly use Express’s `Request` and `Response` types. That’s a practical solution but lambda-expressless doesn’t cover all of the functionalities of Express Request and Response objects yet. And some of the methods have small behaviour differences as I mentioned in Readme file. But the target is to have Request and Response objects as same as ExpressJS. So, I'm not sure about to continue with ExpressJS types directly or try to keep our own custom types for a while.
> 
> Do you have a comment about this? @sanderphilipse

The problem is, on the Readme, we are saying you can use bodyParser.json() as the first argument, but currently on Typescript you can not use because it directly uses express's request and response. Even if our request and response are not same, i believe adding this as a typing will not harm anyone, but will enable typescript users to use bodyparser. I would merge this. :)

**Comment by muratcorlu** [comment]
@cenkingunlugu Actually, in current Readme we never mention TypeScript support. That example is for JS implementation and it's still valid. I'm also eager to continue to use ExpressJS types instead of custom types, but in this case developers that prefers TypeScript will have some UX problems on their editors. It'll look like all of the methods and properties of Express Request and Response objects are available, but that's not correct.

I'm not using TypeScript for my Lambda functions yet and I don't know any other people who is using lambda-expressless with TypeScript, so we can merge this if you are OK with that situation. :)

BTW, if we'll use Express types, we'll also need to remove our Request and Response type definitions to be consistent.

**Comment by sanderphilipse** [comment]
I updated the pull request to include only a typo fix (I'm not sure why that ever worked?) and moving types to DevDependencies now

I think Murat is right and we should wait to change the types until they're actually compatible, which they aren't for now.

File: package.json [hunk 1]
```diff
@@ -20,12 +20,13 @@
   },
   "homepage": "https://github.com/muratcorlu/lambda-expressless#readme",
   "devDependencies": {
+    "@types/aws-lambda": "^8.10.33",
+    "@types/express": "^4.17.2",
```
Comment (muratcorlu) [inline package.json:24]: Then we don't need this for now.

**Comment by muratcorlu** [comment]
I fixed problems about TypeScript and using bodyparser with TypeScript. So I close this PR.

**Pull request closed without merging**